modules/testing/slides.md

```diff
@@ -55,89 +47,104 @@ The loss of the Mars Climate Orbiter in 1999 is another prominent example of a c
 
 ===
 
 <!-- .slide: data-state="standard" -->
 
-## Types of testing
+### Test types
 
-- **Unit tests**: Assess individual components for correctness.
-- **Integration tests**: Verify interactions between different parts of the software.
-- **System tests**: Evaluate the complete and integrated software system.
-- **Regression tests**: Ensure new changes don’t disrupt existing functionality.
+- **Smoke tests** verify that the essential functions of the software work.
+- **Unit tests** assess individual components for correctness.
+- **Integration tests** verify interactions between different parts of the software.
+- **System tests** evaluate the complete and integrated software system.
+- **Regression tests** ensure new changes don’t disrupt existing functionality.
 
 Note:
 
-There are several types of software testing, each of which with its own purpose.
+- Smoke test: A preliminary test to verify that the essential functions of the software work. If a smoke test fails, further testing is halted until the issues are resolved.
+- Unit test: A test that verifies the functionality of an individual unit (e.g., a function, method, or module) in isolation. In unit tests each individual unit of a software is tested separately.
+- Integration test: A test that checks the interaction between integrated components or modules. Individual units (that are tested in isolation using unit tests) are combined and tested as a group. The goal is to detect faults in how these units work together, especially focusing on data flow, APIs, and dependencies.
+- System test: A test that evaluates the complete system as a whole to ensure it meets functional (correct results, no crashes, ...) and non-functional (security, performance, ...) requirements. It checks how different components interact under real-world conditions. It is similar to an integration test but broader in scope.
+- Regression tests: A test that ensures recent code changes haven't negatively affected existing functionality. This often involves re-running previous test cases to confirm that everything still works as expected.
+
+==
 
-In unit tests each individual unit of a software is tested separately. The purpose is to ensure that each component of the software works as expected.
+<!-- .slide: data-state="standard" -->
 
-Integration tests are a level of software testing where individual units are combined and tested as a group. The purpose of this level of testing is to expose faults in the interaction between integrated units.
+### Exercise
 
-The complete, integrated system is checked via the so-called system tests. The purpose here is to evaluate whether the software as a whole gives the correct outputs for given inputs.
+Identify the test types in this story:
```

Comment:
Maybe you should just switch to the exercise tab here? To avoid duplicating the text...